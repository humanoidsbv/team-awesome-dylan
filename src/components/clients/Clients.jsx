import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';

import styles from './Clients.module.css';
import Client from '../client/Client';
import ClientForm from '../client-form/ClientForm';

const Clients = ({ clients, createClient, fetchClients }) => {
  const [isClientFormVisible, setClientFormVisibility] = useState(false);
  const [clientsSort, setClientsSort] = useState(clients);

  const toggleClientForm = () => {
    setClientFormVisibility(!isClientFormVisible);
  };

  useEffect(() => {
    fetchClients();
  }, []);

  return (
    <>
      <ClientForm
        createClient={newClient => createClient(newClient)}
        isClientFormVisible={isClientFormVisible}
        toggleClientForm={toggleClientForm}
      />
      <div className={styles.headerClients}>
        <span className={styles.headerClientsText}> All Clients </span>
        <button
          className={`${styles.newClientButton} ${
            isClientFormVisible ? styles.newClientButtonGrey : ''
          }`}
          disabled={isClientFormVisible}
          onClick={toggleClientForm}
          type="button"
        >
          + New Client
        </button>
        <select type="button" className={styles.sortClientsSelector}>
          <option value="">Sort by:</option>
          {clients.map(({ id, name }) => (
            <option key={id}>{name}</option>
          ))}
        </select>
      </div>
      <div className={styles.clientsContainer}>
        {clients.map(({ id, clientName }) => (
          <Client clientName={clientName} key={id} />
        ))}
      </div>
    </>
  );
};

Clients.propTypes = {
  clients: PropTypes.arrayOf(
    PropTypes.shape({
      clientName: PropTypes.string
    })
  ),
  createClient: PropTypes.func.isRequired,
  fetchClients: PropTypes.func.isRequired
};

Clients.defaultProps = {
  clients: []
};

export default Clients;
