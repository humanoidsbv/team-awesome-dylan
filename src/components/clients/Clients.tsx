import React, { useEffect, useState } from 'react';

import styles from './Clients.module.css';
import Client from '../client/Client';
import ClientForm from '../client-form/ClientForm';
import { ClientsProps, ClientInterface } from '../../redux/clients/types';

const Clients = ({ clients, createClient, fetchClients }: ClientsProps): React.ReactElement => {
  const [isClientFormVisible, setClientFormVisibility] = useState(false);

  const toggleClientForm = (): void => {
    setClientFormVisibility(!isClientFormVisible);
  };

  useEffect(() => {
    fetchClients();
  }, []);

  return (
    <>
      <ClientForm
        createClient={(newClient): {} => createClient(newClient)}
        isClientFormVisible={isClientFormVisible}
        toggleClientForm={toggleClientForm}
      />
      <div className={styles.headerClients}>
        <span className={styles.headerClientsText}> All Clients </span>
        <button
          className={`${styles.newClientButton} ${
            isClientFormVisible ? styles.newClientButtonGrey : ''
          }`}
          disabled={isClientFormVisible}
          onClick={toggleClientForm}
          type="button"
        >
          + New Client
        </button>
        <select className={styles.sortClientsSelector}>
          <option value="">Sort by:</option>
          {clients.map(({ id, clientName }: ClientInterface) => (
            <option key={id}>{clientName}</option>
          ))}
        </select>
      </div>
      <div className={styles.clientsContainer}>
        {clients.map(({ id, clientName }: ClientInterface): React.ReactElement => (
          <Client clientName={clientName} key={id} />
        ))}
      </div>
    </>
  );
};

export default Clients;
