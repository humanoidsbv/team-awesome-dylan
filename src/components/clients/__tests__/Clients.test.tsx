import * as React from 'react';
import { shallow } from 'enzyme';

import Clients from '../Clients';

it('matches snapshot', () => {
  const clients = shallow(
    <Clients
      clients ={[{
        branch:'test',
        clientName:'test',
        id: 123
      }]}
      createClient={({}) => null}
      fetchClients={()=> null}

    />
  );
  expect(clients).toMatchSnapshot();
});
