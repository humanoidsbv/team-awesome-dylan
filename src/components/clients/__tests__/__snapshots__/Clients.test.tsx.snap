// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clients matches snapshot 1`] = `
<Fragment>
  <ClientForm
    createClient={[Function]}
    isClientFormVisible={false}
    toggleClientForm={[Function]}
  />
  <div
    className="headerClients"
  >
    <span
      className="headerClientsText"
    >
       All Clients 
    </span>
    <button
      className="newClientButton "
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      + New Client
    </button>
    <select
      className="sortClientsSelector"
    >
      <option
        value=""
      >
        Sort by:
      </option>
      <option
        key="123"
      >
        test
      </option>
    </select>
  </div>
  <div
    className="clientsContainer"
  >
    <Client
      clientName="test"
      key="123"
    />
  </div>
</Fragment>
`;
